File: Ent_UsAdmin/DescuentoPromesaLegal.vb
Public Class DescuentoPromesaLegal

    Public Maximo As Double
    Public Total As Boolean

    Private Sub DescuentoPromesa_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        lblMaximo.Text = FormatCurrency(Maximo, 2)

    End Sub

    Private Sub txtOtraCantidad_Click(sender As Object, e As EventArgs) Handles txtOtraCantidad.Click
        If txtcantidad.Text > Maximo Then
            MessageBox.Show("No puedes ingresar un descuento mayor al máximo establecido")
        Else
            Me.Close()

        End If
    End Sub

    Private Sub DescuentoPromesa_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing

        If txtcantidad.Text = "" Then


            PromPagoLegal.descuentoTotal = 0
        Else

            If txtcantidad.Text > Maximo Then

                PromPagoLegal.descuentoTotal = 0
            Else

                PromPagoLegal.descuentoTotal = txtcantidad.Text
            End If

        End If



    End Sub



    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub
End Class

File: Ent_UsAdmin/Promesadepago/DescuentoPromesaLegal.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{
    public partial class DescuentoPromesaLegal
    {

        public double Maximo;
        public bool Total;

        public DescuentoPromesaLegal()
        {
            InitializeComponent();
        }

        private void DescuentoPromesa_Load(object sender, EventArgs e)
        {
            lblMaximo.Text = Strings.FormatCurrency(Maximo, 2);

        }

        private void txtOtraCantidad_Click(object sender, EventArgs e)
        {
            if (Conversions.ToDouble(txtcantidad.Text) > Maximo)
            {
                MessageBox.Show("No puedes ingresar un descuento mayor al máximo establecido");
            }
            else
            {
                Close();

            }
        }

        private void DescuentoPromesa_FormClosing(object sender, FormClosingEventArgs e)
        {

            if (string.IsNullOrEmpty(txtcantidad.Text))
            {


                My.MyProject.Forms.PromPagoLegal.descuentoTotal = 0d;
            }

            else if (Conversions.ToDouble(txtcantidad.Text) > Maximo)
            {

                My.MyProject.Forms.PromPagoLegal.descuentoTotal = 0d;
            }
            else
            {

                My.MyProject.Forms.PromPagoLegal.descuentoTotal = Conversions.ToDouble(txtcantidad.Text);

            }



        }



        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }
    }
}